From: Chris Lamb <lamby@debian.org>
Date: Sat, 7 Dec 2019 09:32:20 +0000
Subject: Make the build reproducible

Closes: #946330
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ea25ecf..60c9289 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ db-install: files-install
 db-install-packed:
 	@# Create a compressed tar without gzip timestamp, so tar.gz
 	@# differs only if content is different
-	cd ./usb_modeswitch.d; find * -print0 | LC_ALL=C sort -z | tar --no-recursion --null --files-from=- --mode=go=rX,u+rw,a-s -cf ../configPack.tar
+	cd ./usb_modeswitch.d; find * -print0 | LC_ALL=C sort -z | tar --no-recursion --null --files-from=- --mode=go=rX,u+rw,a-s --owner=0 --group=0 --numeric-owner -cf ../configPack.tar
 	gzip -f9n ./configPack.tar
 	install --mode=644 -t $(PREFIX)/share/usb_modeswitch ./configPack.tar.gz
 	rm -f ./configPack.tar.gz
