Description: Avoids the useless udev rules reload
Author: Didier Raboud <didier@raboud.com>
Origin: vendor
Forwarded: yes (private mail)
Last-Update: 2010-07-18
Index: usb-modeswitch-data.git/Makefile
===================================================================
--- usb-modeswitch-data.git.orig/Makefile	2010-07-18 20:00:44.000000000 +0200
+++ usb-modeswitch-data.git/Makefile	2010-07-18 20:07:48.000000000 +0200
@@ -12,7 +12,7 @@
 
 clean:
 
-install: files-install rules-reload
+install: files-install
 
 files-install:
 	install -d $(ETCDIR)/usb_modeswitch.d
@@ -37,7 +37,7 @@
 		fi \
 	fi
 
-uninstall: files-uninstall rules-reload
+uninstall: files-uninstall
 
 files-uninstall:
 	$(RM) $(RULESDIR)/40-usb_modeswitch.rules
