Description: Avoids the useless udev rules reload
Author: Didier Raboud <didier@raboud.com>
Origin: vendor
Forwarded: yes (private mail)
Last-Update: 2010-06-23
Index: usb-modeswitch-data/Makefile
===================================================================
--- usb-modeswitch-data.orig/Makefile	2010-06-23 11:32:51.000000000 +0200
+++ usb-modeswitch-data/Makefile	2010-06-23 11:33:03.000000000 +0200
@@ -15,7 +15,7 @@
 
 clean:
 
-install: files-install rules-reload
+install: files-install
 
 files-install:
 	install -d $(ETCDIR)/usb_modeswitch.d
@@ -40,7 +40,7 @@
 		fi \
 	fi
 
-uninstall: files-uninstall rules-reload
+uninstall: files-uninstall
 
 files-uninstall:
 	$(RM) $(SBINDIR)/usb_modeswitch
